glitch/analysis/security.py

```diff
@@ -1,15 +1,1107 @@
+                                while resource:
+                                    if (resource.value.lower() in ["*"]) or (":*" in resource.value.lower()):
+                                        wildcarded_resource = True
+                                        break
+                                    i += 1
+                                    resource = self.check_required_attribute(statement.keyvalues, [""], f"resources[{i}]")
+                                if wildcarded_resource:
+                                    errors.append(Error('sec_sensitive_iam_action', resource, file, repr(resource)))
+                elif (element.type in ["resource.aws_iam_role_policy", "resource.aws_iam_policy", 
+                                       "resource.aws_iam_user_policy", "resource.aws_iam_group_policy"]):
+                    policy = self.check_required_attribute(element.attributes, [""], "policy")
+                    if policy:
+                        policy_dict = convert_string_to_dict(policy.value.lower())
+                        if policy_dict and policy_dict["statement"]:
+                            statements = policy_dict["statement"]
+                            if isinstance(statements, dict):
+                                statements = [statements]
+                            for statement in statements:
+                                if statement["effect"] and statement["action"] and statement["resource"]:
+                                    if statement["effect"] == "allow":
+                                        if isinstance(statement["action"], list):
+                                            for action in statement["action"]:
+                                                if ("*" in action):
+                                                    errors.append(Error('sec_sensitive_iam_action', policy, file, repr(policy)))
+                                                    break
+                                        else:
+                                            if ("*" in statement["action"]):
+                                                errors.append(Error('sec_sensitive_iam_action', policy, file, repr(policy)))
+                                        if isinstance(statement["resource"], list):
+                                            for resource in statement["resource"]:
+                                                if (resource in ["*"]) or (":*" in resource):
+                                                    errors.append(Error('sec_sensitive_iam_action', policy, file, repr(policy)))
+                                                    break
+                                        else:
+                                            if (statement["resource"] in ["*"]) or (":*" in statement["resource"]):
+                                                errors.append(Error('sec_sensitive_iam_action', policy, file, repr(policy)))
```

Comment:
This code needs to be simplified in some way, there are too many chained blocks.
It seems to me that this pattern happens a lot in this code:
```
i = 0
action = self.check_required_attribute(statement.keyvalues, [""], f"actions[{i}]")
while action:
    if ("*" in action.value.lower()):
      sensitive_action = True
      break
   i += 1
   action = self.check_required_attribute(statement.keyvalues, [""], f"actions[{i}]")
```
Maybe this could be in a function? 
Also for instance, the check `if policy` could be transformed into `if not policy` with a `return`?

Reply:
Yes I agree. This code was made at the last minute to detect the smell more accurately, I expected it to be provisional and refactored when I had time, and all these checks are necessary. However, it can be simplified as you said, with the changes you suggested.

Reply:
Okok thanks for the clarification!